scipy/optimize/tests/test__basinhopping.py

```diff
@@ -397,20 +398,17 @@ class Test_Metropolis:
     def setup_method(self):
         self.T = 2.
         self.met = Metropolis(self.T)
+        self.res_new = OptimizeResult(success=True, fun=0.)
+        self.res_old = OptimizeResult(success=True, fun=1.)
 
     def test_boolean_return(self):
         # the return must be a bool, else an error will be raised in
         # basinhopping
-        ret = self.met(f_new=0., f_old=1.)
+        ret = self.met(res_new=self.res_new, res_old=self.res_old)
         assert isinstance(ret, bool)
 
     def test_lower_f_accepted(self):
-        assert_(self.met(f_new=0., f_old=1.))
-
-    def test_KeyError(self):
-        # should raise KeyError if kwargs f_old or f_new is not passed
-        assert_raises(KeyError, self.met, f_old=1.)
-        assert_raises(KeyError, self.met, f_new=1.)
```

Comment:
Removed `test_keyError`. This is not publicly documented, and the feature is not used. Why test it?